File: tests/algorithms/x/PHP/financial/straight_line_depreciation.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
function _panic($msg) {
    fwrite(STDERR, strval($msg));
    exit(1);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function straight_line_depreciation($useful_years, $purchase_value, $residual_value) {
  if ($useful_years < 1) {
  _panic('Useful years cannot be less than 1');
}
  if ($purchase_value < 0.0) {
  _panic('Purchase value cannot be less than zero');
}
  if ($purchase_value < $residual_value) {
  _panic('Purchase value cannot be less than residual value');
}
  $depreciable_cost = $purchase_value - $residual_value;
  $annual_expense = $depreciable_cost / (1.0 * $useful_years);
  $expenses = [];
  $accumulated = 0.0;
  $period = 0;
  while ($period < $useful_years) {
  if ($period != $useful_years - 1) {
  $accumulated = $accumulated + $annual_expense;
  $expenses = _append($expenses, $annual_expense);
} else {
  $end_year_expense = $depreciable_cost - $accumulated;
  $expenses = _append($expenses, $end_year_expense);
}
  $period = $period + 1;
};
  return $expenses;
};
  echo rtrim(_str(straight_line_depreciation(10, 1100.0, 100.0))), PHP_EOL;
  echo rtrim(_str(straight_line_depreciation(6, 1250.0, 50.0))), PHP_EOL;
  echo rtrim(_str(straight_line_depreciation(4, 1001.0, 0.0))), PHP_EOL;
  echo rtrim(_str(straight_line_depreciation(11, 380.0, 50.0))), PHP_EOL;
  echo rtrim(_str(straight_line_depreciation(1, 4985.0, 100.0))), PHP_EOL;
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/algorithms/transpiler/FS/financial/straight_line_depreciation.fs
// Generated 2025-08-13 16:13 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let rec _str v =
    let s = sprintf "%A" v
    let s = if s.EndsWith(".0") then s.Substring(0, s.Length - 2) else s
    s.Replace("[|", "[")
     .Replace("|]", "]")
     .Replace("; ", " ")
     .Replace(";", "")
     .Replace("\"", "")
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec straight_line_depreciation (useful_years: int) (purchase_value: float) (residual_value: float) =
    let mutable __ret : float array = Unchecked.defaultof<float array>
    let mutable useful_years = useful_years
    let mutable purchase_value = purchase_value
    let mutable residual_value = residual_value
    try
        if useful_years < 1 then
            ignore (failwith ("Useful years cannot be less than 1"))
        if purchase_value < 0.0 then
            ignore (failwith ("Purchase value cannot be less than zero"))
        if purchase_value < residual_value then
            ignore (failwith ("Purchase value cannot be less than residual value"))
        let depreciable_cost: float = purchase_value - residual_value
        let annual_expense: float = depreciable_cost / (1.0 * (float useful_years))
        let mutable expenses: float array = Array.empty<float>
        let mutable accumulated: float = 0.0
        let mutable period: int = 0
        while period < useful_years do
            if period <> (useful_years - 1) then
                accumulated <- accumulated + annual_expense
                expenses <- Array.append expenses [|annual_expense|]
            else
                let end_year_expense: float = depreciable_cost - accumulated
                expenses <- Array.append expenses [|end_year_expense|]
            period <- period + 1
        __ret <- expenses
        raise Return
        __ret
    with
        | Return -> __ret
ignore (printfn "%s" (_str (straight_line_depreciation (10) (1100.0) (100.0))))
ignore (printfn "%s" (_str (straight_line_depreciation (6) (1250.0) (50.0))))
ignore (printfn "%s" (_str (straight_line_depreciation (4) (1001.0) (0.0))))
ignore (printfn "%s" (_str (straight_line_depreciation (11) (380.0) (50.0))))
ignore (printfn "%s" (_str (straight_line_depreciation (1) (4985.0) (100.0))))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
